src/SWP/Bundle/FixturesBundle/Command/ThemeSetupCommand.php

```diff
@@ -147,21 +157,24 @@ protected function execute(InputInterface $input, OutputInterface $output)
     }
 
     /**
+     * @param ThemeAwareTenantInterface $defaultTenant
+     *
      * @throws \Exception if there is no default tenant
      */
-    private function assignDefaultThemeDefaultTenant()
+    private function assignDefaultTheme(ThemeAwareTenantInterface $defaultTenant)
     {
-        $tenantRepository = $this->getContainer()->get('swp_multi_tenancy.tenant_repository');
-        $defaultTenant = $tenantRepository->findOneBy(['name' => TenantInterface::DEFAULT_TENANT_NAME]);
-        if (null === $defaultTenant) {
-            throw new \Exception('No default tenant found, please first run php app/console swp:tenant:create --default');
-        }
-
         // Only assign default theme if no theme has yet been assigned to default tenant
         if (null === $defaultTenant->getThemeName()) {
             $defaultTenant->setThemeName(self::DEFAULT_THEME_NAME);
-            $em = $this->getContainer()->get('doctrine.orm.entity_manager');
-            $em->flush();
+            $documentManager = $this->getContainer()->get('doctrine_phpcr.odm.document_manager');
+            $documentManager->flush();
+        }
+    }
+
+    private function assertTenantIsFound(ThemeAwareTenantInterface $defaultTenant)
```

Comment:
Yeah, the execute method does two very different things - deleting and setting up a theme. If we're going to put any code that might throw an exception into its own method, we will have to do a lot of refactoring.